pkg/apis/postgres-operator.crunchydata.com/v1beta1/patroni_types.go

```diff
@@ -48,6 +52,21 @@ type PatroniSpec struct {
 	// - https://patroni.readthedocs.io/en/latest/kubernetes.html
 }
 
+type PatroniLogConfig struct {
+
+	// Patroni Log file size
+	// https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity
+	// +required
+	FileSize *string `json:"fileSize"`
+
+	// Patroni log level
+	// https://docs.python.org/3.6/library/logging.html#levels
+	// +kubebuilder:validation:Enum={CRITICAL,ERROR,WARNING,INFO,DEBUG,NOTSET}
```

Comment:
`NOTSET` is a terrible name, but it's official... WDYT of calling it `ALL` here and converting that to `NOTSET` in our code?

Reply:
I was conflicted about this as well but landed on just including all possible options. That said, `NOTSET` is somewhat confusing. Per the [docs I've read](https://docs.python.org/3/library/logging.html#logging-levels), this is the expected behavior:
> When set on a logger, indicates that ancestor loggers are to be consulted to determine the effective level. If that still resolves to NOTSET, then all events are logged. When set on a handler, all events are handled.

If I understand correctly, that means that since we don't have an ancestor loggers, it will always be equivalent to `ALL` logs. If that's the case, I'm good with changing this to `ALL`, I would just want to make sure I'm not missing any other possibilities.

Reply:
👍🏻 That's my understanding as well. This Patroni setting goes into the root logger, so it sounds like "all."

But! This doesn't make sense (to me) in the hierarchy. Does a parent logger set to `DEBUG` have anything to do with debug messages from a child logger set to `INFO`? 🤔 cc: @benjaminjb

I've lost interest in the usability of this field! I think what you have now behaves exactly as Patroni intends. I don't have a better UX for logging than upstream Python. 🤷🏻 👍🏻 🌴